### PR TITLE
WIP: Add Auditing support for App Registrations

### PR DESCRIPTION
**Pull request by sylank**
It is a big changeset because I moved the audit-related services and repositories into another module it is called spring-cloud-dataflow-audit. I wanted to use the audit classes in registry module but I got a cycle dependency error.

* Now the app registration inserts audit logs into the audit table.
* If you create a new module you can use the audit module when you want to report audit records.

this issue:
resolves: #2486
related to: #2456 (because we can create complex audit tests) & #2621 (because I need to move some classes from registry to core module)

### CODE REVIEW

**Comment by sylank** [comment]
I found this issue inside a class in a comment while I was interpreting the code. #2456

Question:
* Some classes (AuditActionType, AuditActionTypeConverter, ...) are placed in the dataflow-core module, should we move into the new audit module? (I do not think so)
* Is make sense to move that classes into that new module?
* Ignore annotation was removed from an H2 test: DataFlowServerConfigurationTests#testNoServer, should I revert it?
* The Flyway scripts have stayed in the server-core module, should I move them into the new module?

**Comment by jvalkeal** [comment]
Thx for the PR! At first look looks relatively ok, we probably didn't thought about usage outside of server core. We don't actually use flyway yet and all those sql scripts will be converted to flyway schemas(I'm currently doing it). Lemmy play with this PR, then I can post more comments.

**Comment by sylank** [comment]
Thank you, I have just added a small unit test and I am waiting for the conclusion :)

File: spring-cloud-dataflow-audit/pom.xml [hunk 1]
```diff
@@ -0,0 +1,117 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (jvalkeal) [inline spring-cloud-dataflow-audit/pom.xml]: This pom needs a cleanup as it was probably just copied over.

File: spring-cloud-dataflow-audit/src/main/java/org/springframework/cloud/dataflow/audit/service/AuditServiceUtils.java [hunk 2]
```diff
@@ -29,21 +33,51 @@
...
 	public static final String COMMANDLINE_ARGUMENTS = "commandlineArguments";
 
 	private final ArgumentSanitizer argumentSanitizer = new ArgumentSanitizer();
 
 	public AuditServiceUtils() {
 	}
 
+	public Map<String, Object> convertAppRegistrationToAuditData(String name, ApplicationType type, String version) {
+		try {
+			return this.convertAppRegistrationToAuditData(new AppRegistration(name, type, version, new URI(""), null));
```
Comment (jvalkeal) [inline spring-cloud-dataflow-audit/src/main/java/org/springframework/cloud/dataflow/audit/service/AuditServiceUtils.java]: I think app `uri` and `metadataUri` are the most important ones which are now not added.

**Comment by jvalkeal** [comment]
Also it's 2019 so maybe use that in headers :)

**Comment by sylank** [comment]
Should I update the year in every modified class or just in the package info files?

**Comment by jvalkeal** [comment]
Well at least those files where you changed some functionality. You can skip files which got changed simply because of a move.

**Comment by jvalkeal** [comment]
There is at least one build failure:
```
[ERROR] Failures: 
[ERROR]   TaskSchedulerControllerTests.testRemoveSchedule:251 expected:<1> but was:<6>
[INFO] 
[ERROR] Tests run: 310, Failures: 1, Errors: 0, Skipped: 1
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for spring-cloud-dataflow-parent 2.0.0.BUILD-SNAPSHOT:
[INFO] 
[INFO] spring-cloud-dataflow-parent ....................... SUCCESS [  1.392 s]
[INFO] spring-cloud-dataflow-configuration-metadata ....... SUCCESS [  4.154 s]
[INFO] spring-cloud-dataflow-core ......................... SUCCESS [  5.775 s]
[INFO] spring-cloud-dataflow-rest-resource ................ SUCCESS [  6.175 s]
[INFO] spring-cloud-dataflow-audit ........................ SUCCESS [  4.300 s]
[INFO] spring-cloud-dataflow-registry ..................... SUCCESS [  4.913 s]
[INFO] spring-cloud-dataflow-completion ................... SUCCESS [  7.012 s]
[INFO] spring-cloud-dataflow-server-core .................. FAILURE [03:16 min]
```

File: pom.xml [hunk 3]
```diff
@@ -84,6 +84,7 @@
 		<module>spring-cloud-dataflow-platform-kubernetes</module>
 		<module>spring-cloud-dataflow-platform-cloudfoundry</module>
 		<module>spring-cloud-starter-dataflow-server-local</module>
+		<module>spring-cloud-dataflow-audit</module>
```
Comment (jvalkeal) [inline pom.xml]: Forget to mention that new module should also go to `spring-cloud-dataflow-dependencies/pom.xml`

Comment (sylank) [reply pom.xml]: In the new commit I added the audit dependenci into your mentioned `spring-cloud-dataflow-dependencies/pom.xml`

**Comment by sylank** [comment]
I have just pushed some changes. I added the audit dependency into the dependencies module. I changed the years too.

**Comment by jvalkeal** [comment]
ok thanks a lot mate. I'll try to merge this today.

**Comment by jvalkeal** [comment]
Merged per 337260317721ea3a1b5a352c3a6721cb020fdc8a

**Pull request closed without merging**